augur/merge.py

```diff
@@ -168,16 +221,102 @@ def run(args: argparse.Namespace):
     metadata: Optional[List[NamedMetadata]] = None
     output_columns: Optional[Columns] = None
     output_source_column: Optional[Callable[[str], str]] = None
+    sequences: Optional[List[Union[NamedSequenceFile, UnnamedSequenceFile]]] = None
+    named_sequences: Optional[List[NamedSequenceFile]] = None
 
     if args.metadata:
         metadata = get_metadata(args.metadata, args.metadata_id_columns, args.metadata_delimiters)
+
+    if args.sequences:
+        sequences = list(get_sequences(args.sequences))
+
+        # Perform checks on file names.
+
+        named_sequences = [s for s in sequences if isinstance(s, NamedSequenceFile)]
+
+        if unnamed_sequences := [s for s in sequences if isinstance(s, UnnamedSequenceFile)]:
+            for x in unnamed_sequences:
+                print_info(f"WARNING: Sequence file {x.path!r} is unnamed. Skipping validation with metadata.")
+
+    if metadata and named_sequences:
+        metadata_names = [m.name for m in metadata]
+        sequences_names = [s.name for s in named_sequences]
+
+        if sequences_missing_metadata := sorted(set(sequences_names) - set(metadata_names)):
+            raise AugurError(dedent(f"""\
+                Named inputs must be paired.
+
+                The following sequence {_n("input does", "inputs do", len(sequences_missing_metadata))} not have a corresponding metadata input:
+
+                  {indented_list([repr(x.path) for x in named_sequences if x.name in sequences_missing_metadata], '                ' + '  ')}
```

Comment:
I'd include the bad inputs as `name=path` in the error message (i.e. not just as `path`), which gives the person staring at the error message two ways to figure out what they're supposed to change.